Update dev-docs (#8730)
numba/numba: docs/source/developer/contributing.rst

build of llvmlite. These are available from the ``numba/label/dev`` channel on
`anaconda.org <https://anaconda.org/numba/llvmlite>`_.

> Then, to create an environment with a few of the most common dependencies. Note
> the use of the double-colon syntax (``numba/label/dev::llvmlite``) to install
> the latest development version of the llvmlite library.

Reviewer: I think "Then" doesn't apply because nothing was done prior. The text also needs a `::` to render the shell command as code:

```suggestion
To create an environment with the required dependencies, noting the use of the 
double-colon syntax (``numba/label/dev::llvmlite``) to install the latest
development version of the llvmlite library::
```

Author: <img width="1124" alt="Screen Shot 2023-05-04 at 10 23 26" src="https://user-images.githubusercontent.com/41563/236149731-c2d78d43-afcc-4992-8e47-45d7a6f7fb1b.png">

Looks good.

''''''''''''''''''

This documentation is under the ``docs`` directory of the `Numba repository`_.
It is built with `Sphinx <http://sphinx-doc.org/>`_, `numpydoc
<https://numpydoc.readthedocs.io/>`_ and the `sphinx-rtd-theme

> <https://sphinx-rtd-theme.readthedocs.io/en/stable/>`.

Reviewer: I think this needs a following underscore to render as a link:

```suggestion
<https://sphinx-rtd-theme.readthedocs.io/en/stable/>_`.
```

Author: <img width="1124" alt="Screen Shot 2023-05-04 at 10 24 17" src="https://user-images.githubusercontent.com/41563/236149951-e082e4ac-ba45-4f00-913e-8142c297826c.png">
 
this one also

Author: The underscore must come after the backtick. I fixed this in 4a030ee85f716e5b2c523ff2f02a57a27560a79b
